app/models/dimensions/item.rb

```diff
@@ -61,18 +64,56 @@ def self.create_empty(content_id, base_path)
     statistical_data_set
     take_part
     topical_event_about_page
+    travel_advice
     working_group
     world_location_news_article
   ].freeze
 
   def extract_by_schema_type(json)
     schema = json.dig("schema_name")
-
     if schema.nil? || !VALID_SCHEMA_TYPES.include?(schema)
       raise InvalidSchemaError, "Schema does not exist"
     end
 
-    html = json.dig("details", "body")
+    case schema
+    when 'licence'
+      html = extract_licence(json)
+    when 'place'
+      html = extract_place(json)
+    when 'guide', 'travel_advice'
+      html = extract_parts(json)
+    else
+      html = extract_main(json)
+    end
+    parse_html(html)
+  end
+
+  def extract_licence(json)
+    json.dig("details", "licence_overview")
+  end
+
+  def extract_place(json)
+    html = []
+    html << json.dig("details", "introduction")
+    html << json.dig("details", "more_information")
+    html.join(" ")
+  end
+
+  def extract_parts(json)
```

Comment:
this logic is really similar to what we use in search for creating the `indexable_content` field. Do you think we should share the code across the apps?

Reply:
Really worth exploring!